onchain/wallet/wallet.go

```diff
@@ -843,3 +880,40 @@ func (wallet *Wallet) Ready() bool {
 func (wallet *Wallet) GetWalletInfo() onchain.WalletInfo {
 	return wallet.WalletInfo
 }
+
+func (wallet *Wallet) estimateFee() (float64, error) {
+	var output Json
+	var estimates struct {
+		Fees []float64 `json:"fees"`
+	}
+	if err := withOutput(C.GA_get_fee_estimates(wallet.session, &output), output, &estimates); err != nil {
+		return 0, err
+	}
+	return estimates.Fees[0] / 1000, nil
+}
+
+func (wallet *Wallet) autoConsolidate() error {
+	if wallet.subaccount != nil {
+		unspent, err := wallet.getUnspentOutputs(*wallet.subaccount, false)
+		if err != nil {
+			return fmt.Errorf("failed to get unspent outputs: %v", err)
+		}
+		for _, outputs := range unspent.Unspent {
+			if len(outputs) > int(config.AutoConsolidateThreshold) {
+				logger.Infof("Auto consolidating %d utxos", len(outputs))
+				address, err := wallet.NewAddress()
+				if err != nil {
+					return fmt.Errorf("new address: %v", err)
+				}
+				feeRate, err := wallet.estimateFee()
+				if err != nil {
+					return fmt.Errorf("could not get estimate fee: %v", err)
+				}
+				if _, err := wallet.SendToAddress(address, 0, feeRate, true); err != nil {
+					return fmt.Errorf("could not send: %v", err)
+				}
```

Comment:
Let's add a info log here. Something along the lines of `Consolidated {len(utxoSpend)} {symbol} UTXOs`